File: erlang/darts/src/darts.erl
-module(darts).

-export([score/2]).


score(0, 0) ->
    10;
score(X, Y) -> 
    case math:sqrt(math:pow(X, 2) + math:pow(Y, 2)) of
        Rad when Rad =< 5.0 ->
            5;
        Rad when Rad >= 5.0 andalso Rad =< 10.0 ->
            1;
        _ ->
            0
    end.

% point system
% 0
% 1
% 5
% 10

% radiuses
% 10
% 5
% 1
% 0 

% Circle
% X(2) + Y(2) = R(2)

File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    shot_radius = :math.sqrt(x ** 2 + y ** 2)

    cond do
      shot_radius <= 1 ->
        10

      shot_radius > 1 and shot_radius <= 5 ->
        5

      shot_radius > 5 and shot_radius <= 10 ->
        1

      shot_radius > 10 ->
        0

      true ->
        0
    end
  end
end
